[RDY] Fix some regressions

tarruda opened the pull request:
Close #3538 #3547

tarruda committed:
os/input.c: Fix assertion in `create_cursorhold_event`

The assertion now considers the case where events are disabled.

oni-link commented on src/nvim/os/input.c:
```diff
@@ -82,9 +82,11 @@ static void cursorhold_event(void **argv)
+  // have been called(inbuf_poll would return kInputAvail)
+  // TODO(tarruda): Cursorhold should be implemented as a timer set during the
+  // `state_check` callback for the states where it can be triggered.
+  assert(!events_enabled || queue_empty(loop.events));
```
Do we still need this `assert()`? If events are enabled we already know that `loop.events` is empty. Or could the call `ui_flush()` in `state_enter()` fill the queue before `os_inchar()` is called?

tarruda replied:
This assert is just to document the state of these variables. I added it because this is a confusing place to implement of cursorhold(But is required for now, cursorhold is still triggered when getting the count in `normal_execute` for example).

When [further refactoring](https://github.com/neovim/neovim/issues/3382#issuecomment-151290282) is performed, I will be able to move cursorhold triggering to the state_enter layer, possibly as a `VimState` flag that enables or disables a cursorhold timer before blocking for keys.

tarruda committed:
tui: Fix abort when stdout and stderr are not tty.

The abort came from using libuv tty handle on non-tty fd. Use uv_pipe_t in these
cases. Also add simple test for this case.

tarruda committed:
tui: Don't use stderr as an output fallback for the TUI

Ref: https://github.com/neovim/neovim/issues/3547#issuecomment-152156639
